rotkehlchen/exchanges/binance.py

```
@@ -1134,8 +1134,8 @@ def _query_online_trade_history(
             iter_markets = list(set(self.selected_pairs).intersection(set(self._symbols_to_pair.keys())))  # noqa: E501
             log.debug(f'Will query the following binance markets: {iter_markets}')
         else:
-            iter_markets = list(self._symbols_to_pair.keys())
-            log.debug('Will query all the binance markets')
+            log.error('No binance markets set. Skipping trade history query.')
+            return []
```

you can change this if/else now


```python
if self.selected_pairs is None:
    log.error('No binance markets set. Skipping trade history query.')
    return []

iter_markets = list(set(self.selected_pairs).intersection(set(self._symbols_to_pair.keys())))  # noqa: E501
log.debug(f'Will query the following binance markets: {iter_markets}')
```

---

this fails silently right? Should we warn the user?